packages/contracts/contracts/BorrowerOperations.sol

```diff
@@ -129,6 +129,20 @@ contract BorrowerOperations is LiquityBase, Ownable, CheckContract, SendCollater
         pcvAddress = _pcvAddress;
         pcv = IPCV(_pcvAddress);
         collateralAddress = _collateralAddress;
+        
+        require(
+            (Ownable(_defaultPoolAddress).owner() != address(0) || 
+            defaultPool.collateralAddress() == _collateralAddress) &&
+            (Ownable(_activePoolAddress).owner() != address(0) || 
+            activePool.collateralAddress() == _collateralAddress) &&
+            (Ownable(_stabilityPoolAddress).owner() != address(0) || 
+            IStabilityPool(stabilityPoolAddress).collateralAddress() == _collateralAddress) &&
+            (Ownable(_collSurplusPoolAddress).owner() != address(0) || 
+            collSurplusPool.collateralAddress() == _collateralAddress) &&
+            (address(pcv.thusdToken()) == address(0) || 
```

Comment:
Maybe dumb question since I didn't quite catch the context but why not

```
(address(pcv.thusdToken()) != address(0) ||
```

(not equal operator)

Reply:
I had next idea in mind: we don't have strict order of deploying and calling setAddresses. So I want to compare collateral addresses between contracts only if that address was already set. And collateral could be zero address. So that check means: if PCV has no set then collateral address is useless. For other contracts I'm using owner field because almost all `setAddress` set owner=zero in the end of the method. So if owner != zero then contract is not set.